github_pr.py

```diff
@@ -108,12 +113,74 @@ def github_list_prs(**args):
     elif 'head' in args and args['head']:
         prs = _load_prs_by_branch(**args)
         _print_prs(prs, **args)
+    elif 'filters' in args and args['filters']:
+        prs = github_filter_prs(**args)
+        _print_prs(prs, **args)
     else:
         prs = repo.get_pulls()
         _print_prs(prs, **args)
     return list_return_obj
 
 
+def github_filter_prs(**args):
+    """
+    Filters prs to return only what is contained in the filters
+    """
+    git_hub = Github(args['token'])
+    repo = git_hub.get_repo(args['repo'])
+    all_prs = [{'pr':repo.get_pull(pr.number), 'issue':repo.get_issue(pr.number)} for pr in repo.get_pulls()]
+    filters = {}
+    for filter_option in args['filters'].split(','):
+        part = filter_option.partition("=")
+        filters[part[0]] = part[2]
+
+    if 'owner' in filters:
+        all_prs = return_specific_owner_prs(all_prs, filters)
+    if 'label' in filters:
+        all_prs = return_specific_labeled_prs(all_prs, filters)
+    if 'status' in filters:
+        all_prs = return_specific_status_prs(all_prs, filters)
+    if 'comment' in filters:
+        all_prs = return_specific_comment_prs(all_prs, filters)
+
+    return all_prs
+
+
+def return_specific_owner_prs(all_prs, filters):
+    return [pr for pr in all_prs if filters['owner'] == pr['pr'].user.login]
+
+def return_specific_labeled_prs(all_prs, filters):
+    pr_issue = []
+    for pr in all_prs:
+        issue = pr['issue']
+        labels = issue.get_labels()
+        matched_labels = [label.name for label in labels if filters['label'] == label.name]
+        if matched_labels:
+            pr_issue.append({'pr':pr['pr'], 'issue':pr['issue']})
+    return pr_issue
+
+
+def return_specific_status_prs(all_prs, filters):
+    pr_issue = []
+    for pr in all_prs:
+        commits = pr['pr'].get_commits()
+        statuses = [[status.state] for status in commits.reversed[0].get_statuses() if filters['status'] == status.state]
+        if statuses:
+            pr_issue.append({'pr':pr['pr'], 'issue':pr['issue']})
+    return pr_issue
+
+
+def return_specific_comment_prs(all_prs, filters):
```

Comment:
these functions should probably be private methods and be explicit that they accept a list of dictionaries of issues and prs.